examples/deadline/EC2-Image-Builder/python/package/lib/deadline_image.py

```diff
@@ -0,0 +1,243 @@
+        """
+        if os_type is OSType.LINUX:
+            s3_uri = f"s3://thinkbox-installers/Deadline/{deadline_version}/Linux/DeadlineClient-{deadline_version}-linux-x64-installer.run"
+        else:
+            s3_uri = f"s3://thinkbox-installers/Deadline/{deadline_version}/Windows/DeadlineClient-{deadline_version}-windows-installer.exe"
+
+        return Asset(
+            scope,
+            f"DeadlineComponentDoc{construct_id}",
+            path= template.template_component(
+                props=template.TemplateProps(
+                    template_path=os.path.join(
+                        os.getcwd(),
+                        "..",
+                        "components",
+                        f"deadline-{os_type.value.lower()}.component.template",
+                    ),
+                    tokens={
+                        "s3uri": s3_uri,
+                        "version": deadline_version,
+                    }
+                )
+            )
+        )
+
+    def create_default_infrastructure_config(self, scope: Construct, construct_id: str) -> CfnInfrastructureConfiguration:
+        """
+        Create the default infrastructure config, which defines the permissions needed by Image Builder during
+        image creation.
+        """
+        image_builder_role_name = f"DeadlineImageBuilderRole{construct_id}"
+        image_builder_role = Role(
+            scope,
+            f"DeadlineImageBuilderRole{construct_id}",
+            assumed_by=ServicePrincipal("ec2.amazonaws.com"),
+            role_name= image_builder_role_name,
+        )
+        image_builder_role.add_to_policy(PolicyStatement(
+            actions=[
+                'ec2messages:AcknowledgeMessage',
+                'ec2messages:DeleteMessage',
+                'ec2messages:FailMessage',
+                'ec2messages:GetEndpoint',
+                'ec2messages:GetMessages',
+                'ec2messages:SendReply',
+                'imagebuilder:GetComponent',
+                's3:Get*',
+                's3:List*',
+                'ssm:DescribeAssociation',
+                'ssm:GetDeployablePatchSnapshotForInstance',
+                'ssm:GetDocument',
+                'ssm:DescribeDocument',
+                'ssm:GetManifest',
+                'ssm:GetParameter',
+                'ssm:GetParameters',
+                'ssm:ListAssociations',
+                'ssm:ListInstanceAssociations',
+                'ssm:PutInventory',
+                'ssm:PutComplianceItems',
+                'ssm:PutConfigurePackageResult',
+                'ssm:UpdateAssociationStatus',
+                'ssm:UpdateInstanceAssociationStatus',
+                'ssm:UpdateInstanceInformation',
+                'ssmmessages:CreateControlChannel',
+                'ssmmessages:CreateDataChannel',
+                'ssmmessages:OpenControlChannel',
+                'ssmmessages:OpenDataChannel',
+            ],
+            resources=['*'],
```

Comment:
Hmm. Could these permissions be restricted any more than just '*' ? 
e.g. the `s3` permissions scoped to a particular bucket, or bucket prefix? ( I think these are for the Deadline installer?)

Reply:
I'm not sure why I didn't use the Amazon managed policies to begin with. I switched to those and added the S3 permissions specifically for our bucket the installers are in.